WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -43,24 +50,66 @@ public SelectCoinsDialogViewModel(WalletViewModel walletViewModel)
 				pocketColumn
 			}
 		};
+
 		Source.SortBy(pocketColumn, ListSortDirection.Descending);
 		Source.RowSelection!.SingleSelect = true;
 
 		SetupCancel(false, true, false);
 		EnableBack = true;
+		NextCommand = ReactiveCommand.Create(() => Close(DialogResultKind.Normal, GetSelectedCoins(items)));
+	}
+
+	private IEnumerable<SmartCoin> GetSelectedCoins(IEnumerable<CoinControlItemViewModelBase> coinControlItemViewModelBases)
+	{
+		return Flatten(coinControlItemViewModelBases)
```

Comment:
Flatten make no sense in general. You will never need the pocket items only the coins.

```c#
private static IEnumerable<CoinCoinControlItemViewModel> GetAllCoinChildren(IEnumerable<CoinControlItemViewModelBase> r) =>
		r.SelectMany(x => x.Children).OfType<CoinCoinControlItemViewModel>();
```

method should be enough for every purpose.

Reply:
Done! But I've called the method `GetAllCoins`.